api/token.go

```diff
@@ -58,10 +61,29 @@ func (a *API) ResourceOwnerPasswordGrant(ctx context.Context, w http.ResponseWri
 		return oauthError("invalid_grant", "Email not confirmed")
 	}
 
+	// User was previously locked out
+	if lock.Enabled && user.IsLocked(lock.Duration) {
+		return oauthError("invalid_grant", fmt.Sprintf("Account locked for %d minutes", lock.Duration))
+	}
+
 	if !user.Authenticate(password) {
+		if lock.Enabled {
+			user.FailedSignIn(lock.MaxSignInAttempts)
+			// User is now officially locked out of their account
+			if err = a.db.UpdateUser(user); err != nil && user.IsLocked(lock.Duration) {
```

Comment:
hmmm might be worth having this be separate branches for the case that there is an error. Otherwise, you're going to swallow all the errors that come from updating a user. 

And the condition seems wrong, `err != nil` would indicate that we _failed_ to update that user record.